src/ClientApp/src/components/MainMap.js

```diff
@@ -1,93 +1,177 @@
+import { ZoomToExtent, defaults as defaultControls } from "ol/control";
+import { ZoomToLatest } from "./ZoomToLatestControl";
+import { InfoButton } from "./InfoButtonControl";
+import Popup from "./Popup";
 import "ol/ol.css";
 
 export default function MainMap(props) {
   const { standorte } = props;
   const [map, setMap] = useState();
   const [bohrungenLayer, setBohrungenLayer] = useState();
-  const [kantonsgrenze, setKantonsgrenze] = useState();
-  const [kantonsgrenzeLayer, setKantonsgrenzeLayer] = useState();
+  const [latestExtent, setLatestExtent] = useState();
+  const [doZoom, setDoZoom] = useState(true);
+  const [selectedFeature, setSelectedFeature] = useState();
+  const [showInfo, setShowInfo] = useState(false);
+  const [popupVisible, setPopupVisible] = useState(false);
+  const [popup, setPopup] = useState();
 
   const mapElement = useRef();
-  const mapRef = useRef();
-  mapRef.current = map;
-  mapElement.current = map;
+  const popupElement = useRef();
 
-  // Get Kantonsgrenze
-  useEffect(() => {
-    fetch(
-      "https://api3.geo.admin.ch/rest/services/api/MapServer/ch.swisstopo.swissboundaries3d-kanton-flaeche.fill/11?geometryFormat=geojson&sr=2056"
-    )
-      .then((response) => response.json())
-      .then((fetchedFeatures) => {
-        setKantonsgrenze([new GeoJSON().readFeature(fetchedFeatures.feature)]);
+  const handleZoomToLatestExtend = () => setDoZoom(true);
+  const resetZoom = () => setDoZoom(false);
+  const handleInfoClick = () => setShowInfo(true);
+  const closePopup = () => setPopupVisible(false);
+
+  const defaultStyle = new Style({
+    image: new Circle({
+      radius: 4,
+      stroke: new Stroke({
+        color: [25, 118, 210, 1],
+        width: 2,
+      }),
+      fill: new Fill({
+        color: [25, 118, 210, 0.3],
+      }),
+    }),
+  });
+
+  const selectedStyleFunction = (feature) => {
+    if (selectedFeature !== feature) {
+      setSelectedFeature(feature);
+      return new Style({
+        image: new Circle({
+          radius: 5,
+          stroke: new Stroke({
+            color: [233, 197, 19, 1],
+            width: 5,
+          }),
+          fill: new Fill({
+            color: [25, 118, 210, 0.3],
+          }),
+        }),
       });
-  }, []);
+    } else {
+      return defaultStyle;
+    }
+  };
 
   // Initialize map on first render
   useEffect(() => {
     // Add custom projection for LV95
     const projection = new Projection({
       code: "EPSG:2056",
-      extent: [2485071.58, 1075346.31, 2828515.82, 1299941.79],
+      extent: [2572670, 1211017, 2664529, 1263980],
       units: "m",
     });
     addProjection(projection);
 
-    const landeskarte = new TileLayer({
-      source: new TileWMS({
-        url: "https://wms.geo.admin.ch/?VERSION=1.3.0&lang=de",
-        params: { LAYERS: "ch.swisstopo.pixelkarte-grau", TILED: true },
-        serverType: "mapserver",
-        projection: projection,
-        crossOrigin: "anonymous",
+    //Add controls
```

Comment:
PP: Durchgängig immer einen Whitespace zwischen `//` und dem Kommentar setzen
```suggestion
    // Add controls
```